### PR TITLE
Updating Button stories for Icon only testing

### PR DESCRIPTION
**Pull request by LFDanLu**
Closes 

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

Test the Button stories with controls

## 🧢 Your Project:

RSP

### CODE REVIEW

File: packages/@react-spectrum/button/stories/Button.stories.tsx [hunk 1]
```diff
@@ -105,36 +73,44 @@ storiesOf('Button', module)
...
-        isQuiet
-        {...props}>
-        Quiet
-      </Button>
+        <Button {...buttonProps} isQuiet>
+          {showIconWithText && <Bell />}
+          <Text>Quiet</Text>
+        </Button>
       )}
+      <Button {...buttonProps}>
```
Comment (devongovett) [inline packages/@react-spectrum/button/stories/Button.stories.tsx:99]: Can you add an `aria-label` here? I see an error in the accessibility tab:

image

File: packages/@react-spectrum/button/stories/Button.stories.tsx [hunk 2]
```diff
@@ -105,36 +73,44 @@ storiesOf('Button', module)
     () => renderStyles()
   );
 
-function render<T extends ElementType = 'button'>(props: SpectrumButtonProps<T> = {variant: 'primary'}) {
-  return (
+function render<T extends ElementType = 'button'>(props: SpectrumButtonProps<T> & {showIconWithText?: boolean} = {variant: 'primary'}) {
+  let {
+    showIconWithText,
```
Comment (devongovett) [inline packages/@react-spectrum/button/stories/Button.stories.tsx:78]: Does this need to be a control? Should we just render one with and one without always?

**Review by snowystinger — DISMISSED** [review]

File: packages/@react-spectrum/button/stories/Button.stories.tsx [hunk 3]
```diff
@@ -106,35 +78,93 @@ storiesOf('Button', module)
...
+
+function renderIconOnly<T extends ElementType = 'button'>(props: SpectrumButtonProps<T> = {variant: 'primary'}) {
+  let buttonProps = mergeProps(props, actions);
+
+  let buttons = (
+    <Flex gap="size-200">
+      <Button {...buttonProps} aria-label="default icon only button">
+        <Bell />
+      </Button>
+      {props.variant !== 'cta' && (
```
Comment (ktabors) [inline packages/@react-spectrum/button/stories/Button.stories.tsx]: Please add a disabled prop button to this story.

**Review by ktabors — COMMENTED** [review]
I like that this reduced the number of stories. I think there is a way to disable a control for a story and we should probably do that for the last two.

**Comment by LFDanLu** [comment]
> I think there is a way to disable a control for a story and we should probably do that for the last two.

Did a brief search for how to do this for the storiesOf format, but only found stuff for the newer formats... Lemme know if you know how to do this but otherwise gonna punt on this for now since I'd like to get this in for testing tmrw

**Review by reidbarber — APPROVED** [review]

**Review by snowystinger — APPROVED** [review]

**Pull request merged**